python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -609,12 +613,18 @@ def get_runs_feed_entries(
     else:
         backfills = []
 
-    runs = [
-        GrapheneRun(run)
-        for run in instance.get_run_records(
-            limit=fetch_limit, cursor=runs_feed_cursor.run_cursor, filters=run_filters
-        )
-    ]
+    # if we are not showing runs within backfills and the backfill_id filter is set, we know
+    # there will be no results, so we can skip fetching runs
+    should_fetch_runs = not (exclude_subruns and _filters_contain_backfill_id(run_filters))
```

Comment:
when we add the option to only show backfills this condition will become more complicated and probably warrant it's own function, but seems overkill for now since it's still unclear how the FE will work for only showing backfills so i dont want to make assumptions now on how we will support that feature